STM32Resources/BuildEnvironment_Setup/BuildEnvironment_Setup.adoc

```diff
@@ -0,0 +1,81 @@
+
+
+= *Setting up and Using an STM32 Build Environment on Linux*
+A guide for setting up a linux system to build projects for STM32
+
+
+This guide assumes you are starting with a Windows 64 bit system and intend to use a virtual machine in Oracle Virtualbox. I installed Linux Mint 18.3 If you already have a linux system set up, are using a Mac, or want to set it up a diffeent way figure out what parts of this guide you have already completed yourself and rejoin at a later step.
+
+== Install Virtualbox
+Download https://www.virtualbox.org/wiki/Downloads[Oracle Virtual Box]
+The latest version at writting was 5.2.4. Also download the VirtualBox Extension Pack from the same page.
+Install both.
+
+==== Setup a Virtual Machine
+Create a new virtual machine in Virtualbox.
+
+image::https://github.com/StudentOrganisationForAerospaceResearch/DevelopmentResources/blob/garnettanner/stm32-example/STM32Resources/BuildEnvironment_Setup/images/1.PNG[]
+
+There isn't an option or Linux Mint OS so choose Ubuntu.
+You'll need 20-30+GB of hard disk space to hold the OS, tools and your projects.
+Run through the creation of your VM and hard disk configuration
+
+
+
+==== Install Linux Mint
+Download the https://linuxmint.com/edition.php?id=246[Linux Mint Cinnamon 64 bit ISO]
+When Virtualbox asks to select a start up disk, navigate to the ISO which will boot into the installer.
+Click the disk icon to start the installation and run through it.
+
+image::https://github.com/StudentOrganisationForAerospaceResearch/DevelopmentResources/blob/garnettanner/stm32-example/STM32Resources/BuildEnvironment_Setup/images/2.PNG[]
+
+image::https://github.com/StudentOrganisationForAerospaceResearch/DevelopmentResources/blob/garnettanner/stm32-example/STM32Resources/BuildEnvironment_Setup/images/3.PNG[]
+
+==== Configure Virtual Machine
+You'll need to enable a fe
+w features in Virtualbox to allow everything to work.
+One is the Guest Additions Package. At the top of your VM window select Devices->Insert Guest Additions CD Image. Run the installer.
+
+You should now be able to use shared clipboard  between the host and guest operating systems once you enable these features in the Devices menu.
+
+Shut down your VM. Go into settings and enable 3D acceleration for the display (if available) and the USB controller. When you plug in a USB device, you can capture it in the VM by clicking the USB icon in the bottom right of the window and selecting your device.
+
+==== Install Git and other Packages
+==== Git
+----
+ sudo apt-get install git-all
+----
+
+==== STLINK, CubeMXMakefile
+Clone the Development Resources repo on the SOAR Git. Open the terminal and navigate
```

Comment:
`on the SOAR Git` is kinda of awkward sounding. I think `...from the SOAR Github Organization` would be better.